test/fixtures/es-module-loaders/assertionless-json-import.mjs

```diff
@@ -1,17 +1,21 @@
 const DATA_URL_PATTERN = /^data:application\/json(?:[^,]*?)(;base64)?,([\s\S]*)$/;
-const JSON_URL_PATTERN = /\.json(\?[^#]*)?(#.*)?$/;
+const JSON_URL_PATTERN = /^[^?]+\.json(\?[^#]*)?(#.*)?$/;
 
-export function resolve(url, context, next) {
-  // Mutation from resolve hook should be discarded.
-  context.importAssertions.type = 'whatever';
-  return next(url);
-}
+export async function resolve(url, context, next) {
+  const { importAssertions: { type } } = context;
 
-export function load(url, context, next) {
-  if (context.importAssertions.type == null &&
+  if (type == null &&
       (DATA_URL_PATTERN.test(url) || JSON_URL_PATTERN.test(url))) {
```

Comment:
Did you see https://github.com/nodejs/node/pull/46153#discussion_r1066211657? This is not correct and won't work with most specifiers.

Reply:
I’m not sure what I’m looking at. The fixture doesn’t currently have a `load` hook. And why would the specifier/URL distinction matter when comparing `context.importAssertions.type` versus `type`?

Reply:
`url` here is a specifier string. `result.url` is a URL. That's the key difference I'm talking about.